testsuite/tests/typing-local/local.ml

```diff
@@ -571,6 +577,52 @@ Line 3, characters 63-64:
                                                                    ^
 Error: The value a is local, so cannot be used inside a closure that might escape
 |}]
+let overapp ~(local_ a) ~b = (); fun ~c ~d -> ()
+
+let () = overapp ~a:1 ~b:2 ~c:3 ~d:4
+[%%expect{|
+val overapp : a:local_ 'a -> b:'b -> (c:'c -> d:'d -> unit) = <fun>
+Line 3, characters 9-26:
+3 | let () = overapp ~a:1 ~b:2 ~c:3 ~d:4
+             ^^^^^^^^^^^^^^^^^
+Error: This application is complete, but surplus arguments were provided afterwards.
+       When passing or calling a local value, extra arguments are passed in a separate application.
+  Hint: Try wrapping the marked application in parentheses.
+|}]
+
+let () = overapp ~b:2 ~a:1 ~c:3 ~d:4
+[%%expect{|
+Line 1, characters 20-21:
+1 | let () = overapp ~b:2 ~a:1 ~c:3 ~d:4
+                        ^
+Error: This application is complete, but surplus arguments were provided afterwards.
+       When passing or calling a local value, extra arguments are passed in a separate application.
+  Hint: Try splitting the application in two, first applying the arguments
+  up to the marked one (in the function's type), and then the rest.
```

Comment:
The wording on this hint is particularly awkward. How about:
```
  Hint: Try splitting the function application in two. The arguments that come after this
  one in the function's type should be applied separately.
```

Reply:
Yeah, that's obviously much better.